scripts/upgrade.sh

```
@@ -0,0 +1,127 @@
+        echo "⚠️  Warning: Could not find upstream commit hash in last upgrade message. Using commit date instead."
+        SINCE_DATE="--since=\"$LAST_UPGRADE_DATE\""
+    else
+        echo "📍 Last upstream commit: $UPSTREAM_COMMIT"
+        SINCE_COMMIT="$UPSTREAM_COMMIT"
+    fi
+else
+    echo "📅 No previous upgrade commits found. Looking for initial commit..."
+    INIT_COMMIT=$(git log --oneline --reverse | head -n 1 | cut -d' ' -f1)
+    INIT_DATE=$(git show -s --format="%ci" "$INIT_COMMIT")
+    echo "📅 Initial commit: $INIT_COMMIT ($INIT_DATE)"
+    SINCE_DATE="--since=\"$INIT_DATE\""
```

Shellcheck (0.10.0)

[warning] Quotes/backslashes will be treated literally. Use an array.

(SC2089)